sdk/instance/instance.go

```diff
@@ -102,27 +102,25 @@ func (i *Instance) Create(serviceHash hash.Hash, env []string) (*instance.Instan
 	if err != nil {
 		return nil, err
 	}
+
 	// calculate the final env vars by overwriting user defined one's with defaults.
 	instanceEnv := xos.EnvMergeMaps(xos.EnvSliceToMap(srv.Configuration.Env), xos.EnvSliceToMap(env))
 
 	// calculate instance's hash.
-	h := hash.New()
-	h.Write(srv.Hash)
-	h.Write([]byte(xos.EnvMapToString(instanceEnv)))
-	instanceHash := h.Sum(nil)
+	inst := &instance.Instance{
+		ServiceHash: srv.Hash,
+		EnvHash:     hash.Dump(instanceEnv),
```

Comment:
Does the hash is the same when the order of the elements of `instanceEnv` is not the same?
The function `xos.EnvMapToSlice` or `xos. EnvMapToString` could be used as it sorts the elements.

Reply:
And of course with test please!

Reply:
The env should be sorted in order to create a deterministic hash
```suggestion
		EnvHash:     hash.Dump(sort.Stable(sort.StringSlice(instanceEnv))),
```

Reply:
It's already done in xos Package